compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1701,8 +1701,10 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
         {
             for &local_def_id in tcx.mir_keys(()) {
                 if let DefKind::AssocFn | DefKind::Fn = tcx.def_kind(local_def_id) {
-                    record_array!(self.tables.deduced_param_attrs[local_def_id.to_def_id()] <-
-                        self.tcx.deduced_param_attrs(local_def_id.to_def_id()));
+                    if tcx.intrinsic(local_def_id).map_or(true, |i| !i.must_be_overridden) {
```

Comment:
Does using a new `fake_body` or whatever intrinsic not help?

Reply:
it does, but then I lose my nice assertion 😆

Reply:
I feel like the assertion is less important than the maintenance cost of not sprinkling random checks like this in random places. If something does break and it chooses the runtime body it will crash immediately at runtime, which seems okay. But I'm not sure.